### DISCUSSION TITLE
Feature: Add support for multiple user inputs as part of typer.testing

### DISCUSSION THREAD

**Discussion by apizz** [fastapi/typer#932] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import os
import sys

import pytest
from typer.testing import CliRunner

sys.path.append(os.path.join(os.path.dirname(__file__), "../scripts"))
from main import app

runner = CliRunner()

GROUP_NAME2 = "a-test-group"

def test_group_delete1() -> None:
    """Delete a test group which is a member of another group"""
    result = runner.invoke(app, ["group", "delete", GROUP_NAME2], input=["y", "y"])
    assert result.exit_code == 0
    assert "SUCCESS" in result.stdout
    assert "Delete group" in result.stdout
    assert "Really delete" in result.stdout
```

### Description

I have a script which has a function for deleting a group: `cli group delete <group_name>`.  Given delete actions are destructive, there is a need to require user input to verify the user does indeed wish to delete the group.  However, groups can be a part of one or more other groups (nested groups) and can have more disruptive consequences if deleted without first addressing those relationships.

Rather than producing a different single message that the user is aware the group is a part of another group and to proceed with deletion, the CLI tool prompts the user a second time for input to ensure they really do want to delete the group (and by extension accept the consequents if they opt to move forward).

As a result of this, when attempting to write a valid typer test for this, I run into errors.  It would be nice if the `input` variable within runner.invoke() could accept either a single string (as it is now) _or an array of strings_ so that for other tools that have many user prompts in one command that these can be processed in sequence.

Thank you in advance for the consideration.

### Operating System

macOS

### Operating System Details

Sonoma 14.6

### Typer Version

0.12.3

### Python Version

Python 3.11.9

**Comment by YuriiMotov** [comment, marked as answer]
You can pass multiple inputs in one string with `\n` separator:
```py
    result = runner.invoke(app, ["group", "delete", GROUP_NAME2], input=("y\n" + "y\n"))
```

Full code:

```py
import typer
from typer.testing import CliRunner

app = typer.Typer()


@app.command()
def delete(group: str):
    typer.confirm("Are you sure?", abort=True)
    if True:  # Group has sub-groups
        typer.confirm("This group has sub-groups. Are you really sure?", abort=True)
        typer.echo("Ok, deleted.")


@app.command()
def add(group: str):
    pass


if __name__ == "__main__":
    app()


def test_():
    runner = CliRunner()
    res = runner.invoke(app, ["delete", "GROUP_NAME2"], input=("y\n" + "y\n"))
    assert "Ok, deleted." in res.output
```
